BasicClasses/BankAccount.cs

```diff
@@ -128,5 +128,30 @@ public BankAccount(BankAccountType type)
             _ClosedNumber++;
             Type = type;
         }
+
+        public static bool operator ==(BankAccount acc1, BankAccount acc2)
+        {
+            return (acc1.Type == acc2.Type);
+        }
+
+        public static bool operator !=(BankAccount acc1, BankAccount acc2)
+        {
+            return !(acc1.Type == acc2.Type);
+        }
+
+        public bool Equals(BankAccount acc2)
+        {
+            return (this.Type == acc2.Type);
```

Comment:
А если acc2 будет null? То Вас уволят с работы.
```suggestion
            return !ReferenceEquals(null, acc2) && Number == acc2.Number;
```